### PR TITLE
Adding json formatiing in pre-commit hook

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR adds JSON formatting in the pre-commit hook and Makefile

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .githooks/pre-commit [hunk 1]
```diff
@@ -30,5 +30,13 @@ for FILE in ${STAGED_SH_FILES}; do
 	git add "${FILE}"
 done
 
+echo "==> Formatting changed JSON files..."
+STAGED_JSON_FILES=$(git diff --cached --name-only | grep ".json$")
+for FILE in ${STAGED_JSON_FILES}; do
+  prettyFile=$(jq . "${FILE}")
+	echo "${prettyFile}" > "${FILE}"
+	git add "${FILE}"
+done
```
Comment (andreaangiolillo) [inline .githooks/pre-commit:39]: Main change

**Review by colm-quinn — APPROVED** [review]
LGTM, thank you for the cleanup!

File: .githooks/pre-commit [hunk 2]
```diff
@@ -30,5 +30,13 @@ for FILE in ${STAGED_SH_FILES}; do
 	git add "${FILE}"
 done
 
+echo "==> Formatting changed JSON files..."
+STAGED_JSON_FILES=$(git diff --cached --name-only | grep ".json$")
+for FILE in ${STAGED_JSON_FILES}; do
+  prettyFile=$(jq . "${FILE}")
```
Comment (colm-quinn) [inline .githooks/pre-commit:36]: [nit] In the future, we should consider moving away from a jq dep to something we can guarantee is in the system path.

Slight concern our code is brittle with external dependences that the pre-commit hook can't ensure are present.

Comment (andreaangiolillo) [reply .githooks/pre-commit:36]: Right now we are installing the devtools used in the hook [here](https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/e1be514ee2cf1bd444f236479d504c4bc65c5810/Makefile#L23). Updated the make file to install `jq` using brew

Comment (andreaangiolillo) [reply .githooks/pre-commit:36]: reverted my changes as using brew makes it hard to install devtools in github action. I will create a ticket to migrate away from jq

Comment (colm-quinn) [reply .githooks/pre-commit:36]: That then adds a dep on brew, which may not be on the build env. Don't think that's better :)

Comment (andreaangiolillo) [reply .githooks/pre-commit:36]: https://jira.mongodb.org/browse/INTMDB-615

**Pull request merged**